### PR TITLE
Refactor and add method for plugins to reply to requests

### PR DESCRIPTION
**Pull request by hbina**
Libraries should never panic.

### CODE REVIEW

**Comment by panekj** [comment]
If we're going the route of "libraries should ..." then you should remove specific versions to dependencies

**Comment by hbina** [comment]
Well I have done as you said but this is definitely broken because `lapce` is still locked to the old `psp-types`.

```
hbina@akarin ~/g/lapce (master)> cargo tree -p psp-types
psp-types v0.1.0 (https://github.com/lapce/psp-types#2513ff3c)
```

File: Cargo.toml [hunk 1]
```diff
@@ -6,14 +6,15 @@ description = "rust libary for Lapce plugin"
...
-anyhow = "1.0"
 bytes = "1"
 http = "0.2"
 wasi-experimental-http = { git = "https://github.com/lapce/wasi-experimental-http" }
 once_cell = "1.13.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0.59"
 jsonrpc-lite = "0.5.0"
+thiserror = "1.0.38"
+anyhow = "1.0.68"
```
Comment (panekj) [inline Cargo.toml]: dependencies still point to specific versions instead of major ones

**Comment by panekj** [comment]
> Well I have done as you said but this is definitely broken because `lapce` is still locked to the old `psp-types`.
> 
> ```
> hbina@akarin ~/g/lapce (master)> cargo tree -p psp-types
> psp-types v0.1.0 (https://github.com/lapce/psp-types#2513ff3c)
> ```

What is broken?

File: Cargo.toml (hunk 1 above)
Comment (hbina) [reply Cargo.toml]: Oh you meant that

File: Cargo.toml [hunk 2]
```diff
@@ -6,14 +6,15 @@ description = "rust libary for Lapce plugin"
 edition = "2021"
 
 [dependencies]
-anyhow = "1.0"
 bytes = "1"
 http = "0.2"
 wasi-experimental-http = { git = "https://github.com/lapce/wasi-experimental-http" }
 once_cell = "1.13.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0.59"
```
Comment (panekj) [inline Cargo.toml]: you could do that for `serde_json` as well

File: Cargo.toml [hunk 3]
```diff
@@ -6,14 +6,15 @@ description = "rust libary for Lapce plugin"
 edition = "2021"
 
 [dependencies]
-anyhow = "1.0"
 bytes = "1"
 http = "0.2"
 wasi-experimental-http = { git = "https://github.com/lapce/wasi-experimental-http" }
 once_cell = "1.13.0"
```
Comment (panekj) [inline Cargo.toml]: and `once_cell`

**Comment by hbina** [comment]
>What is broken?

Since newer commits in `psp-types` in master contain changes to a struct and this is not locked to a specific version. So any plugin that depend on it will grab the latest one. So plugins compiled with this version will not work because of serialization/deserialization error from the plugin side.

**Pull request merged**